python/cudaq/__init__.py

```diff
@@ -39,3 +42,82 @@
     initKwargs['target'] = sys.argv[sys.argv.index('--target') + 1]
 
 initialize_cudaq(**initKwargs)
+
+# Expose global static quantum operations
+h = h()
+x = x()
+y = y()
+z = z()
+s = s()
+t = t()
+
+rx = rx()
+ry = ry()
+rz = rz()
+r1 = r1()
+swap = swap()
+
+
+class MidCircuitMeasurementAnalyzer(ast.NodeVisitor):
+    """The `MidCircuitMeasurementAnalyzer` is a utility class searches for 
+       common measurement - conditional patterns to indicate to the runtime 
+       that we have a circuit with mid-circuit measurement and subsequent conditional 
+       quantum operation application."""
+
+    def __init__(self):
+        self.measureResultsVars = []
+        self.hasMidCircuitMeasures = False
+
+    def visit_Assign(self, node):
+        target = node.targets[0]
+        if not 'func' in node.value.__dict__:
+            return
+        creatorFunc = node.value.func
+        if 'id' in creatorFunc.__dict__ and creatorFunc.id in [
+                'mx', 'my', 'mz'
+        ]:
+            self.measureResultsVars.append(target.id)
+
+    def visit_If(self, node):
+        condition = node.test
+        if 'id' in condition.__dict__ and condition.id in self.measureResultsVars:
+            self.hasMidCircuitMeasures = True
+        elif isinstance(condition,
+                        ast.Call) and condition.__dict__['func'].id in [
+                            'mx', 'my', 'mz'
+                        ]:
+            self.hasMidCircuitMeasures = True
+        elif isinstance(condition, ast.UnaryOp):
+            operand = condition.operand
```

Comment:
Is UnaryOp for `not`? Should we think about expressions like `if mz(q1) or mz(q2):`

Reply:
Yes, I will add this in the next commit.